pkg/xray/reports.go

```diff
@@ -0,0 +1,802 @@
+package xray
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+var getReportSchema = func(filtersSchema map[string]*schema.Schema) map[string]*schema.Schema {
+	return util.MergeMaps(
+		getProjectKeySchema(false, ""),
+		map[string]*schema.Schema{
+			"report_id": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				Computed:         true,
+				Description:      "Report ID",
+				ValidateDiagFunc: validator.StringIsNotEmpty,
+			},
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				Description:      "Name of the report.",
+				ValidateDiagFunc: validator.StringIsNotEmpty,
```

Comment:
Apart from not empty, is there any other validations we should apply here? Valid characters, length, etc.

Reply:
Xray documentation doesn't provide such details.